identity-service/src/audiusLibsInstance.js

```diff
@@ -3,36 +3,45 @@ const AudiusLibs = require('@audius/libs')
 const config = require('./config')
 const registryAddress = config.get('registryAddress')
 const web3ProviderUrl = config.get('web3Provider')
+class AudiusLibsWrapper {
+  constructor () {
+    this.audiusLibsInstance = null
+  }
 
-async function initAudiusLibs () {
-  const dataWeb3 = await AudiusLibs.Utils.configureWeb3(web3ProviderUrl, null, false)
-  if (!dataWeb3) throw new Error('Web3 incorrectly configured')
+  async init () {
+    const dataWeb3 = await AudiusLibs.Utils.configureWeb3(web3ProviderUrl, null, false)
+    if (!dataWeb3) throw new Error('Web3 incorrectly configured')
 
-  let audiusInstance = new AudiusLibs({
-    discoveryProviderConfig: AudiusLibs.configDiscoveryProvider(
-      /** whitelist */ new Set([config.get('notificationDiscoveryProvider')])
-    ),
-    ethWeb3Config: AudiusLibs.configEthWeb3(
-      config.get('ethTokenAddress'),
-      config.get('ethRegistryAddress'),
-      config.get('ethProviderUrl'),
-      config.get('ethOwnerWallet')
-    ),
-    web3Config: {
-      registryAddress,
-      useExternalWeb3: true,
-      externalWeb3Config: {
-        web3: dataWeb3,
-        // this is a stopgap since libs external web3 init requires an ownerWallet
-        // this is never actually used in the service's libs calls
-        ownerWallet: config.get('relayerPublicKey')
-      }
-    },
-    isServer: true
-  })
+    let audiusInstance = new AudiusLibs({
+      discoveryProviderConfig: AudiusLibs.configDiscoveryProvider(),
+      ethWeb3Config: AudiusLibs.configEthWeb3(
+        config.get('ethTokenAddress'),
+        config.get('ethRegistryAddress'),
+        config.get('ethProviderUrl'),
+        config.get('ethOwnerWallet')
+      ),
+      web3Config: {
+        registryAddress,
+        useExternalWeb3: true,
+        externalWeb3Config: {
+          web3: dataWeb3,
+          // this is a stopgap since libs external web3 init requires an ownerWallet
+          // this is never actually used in the service's libs calls
+          ownerWallet: config.get('relayerPublicKey')
+        }
+      },
+      isServer: true
+    })
 
-  await audiusInstance.init()
-  return audiusInstance
+    await audiusInstance.init()
+    this.audiusLibsInstance = audiusInstance
+  }
+
+  audiusLibs () {
```

Comment:
@cheran-senthil super nit - can we rename this getAudiusLibs()? that's the last ask i have i promise

Reply:
Not at all! I forgot this while doing it in a rush, fixed 👍